feat : 엔티티 생성 & Dto 생성 & MealServiceImpl에 페이징처리 & 스케줄러

Hue-Jo opened the pull request:
### 변경사항
<!-- 이 PR에서 어떤점들이 변경되었는지 기술해주세요. 가급적이면 as-is, to-be를 활용해서 작성해주세요.  -->
**AS-IS**
* 엔티티 생성
* Dto 생성
* MealServiceImpl 페이징처리

**TO-BE**
* 이전 pr에서 조언해주신대로 String.class -> MealResponse.class로 바로 넣어보려했으나 500에러가 발생합니다. 문제가 어디서 발생하는지를 모르겠습니다. 일단은 String.class로 넣어 진행하다가 방법을 찾으면 수정하겠습니다. 
 
### 테스트
<!-- 본 변경사항이 테스트가 되었는지 기술해주세요 --> 
- [ ] 테스트 코드
- [ ] API 테스트

Hue-Jo committed:
Create Entity And Dto

엔티티 생성 & Dto 생성
MealServiceImpl에 페이징처리

Hue-Jo committed:
Create Scheduler with SpringBatch

mingming-mentor commented on SchoolMealBoard/src/main/java/com/zerobase/schoolmealboard/service/impl/MealServiceImpl.java:
```diff
@@ -53,75 +53,97 @@ public void fetchAndSaveMealInfo() {
+              LocalDate mealDate = LocalDate.parse(row.getMealDate(), DATE_FORMAT);
+
+              // 기존 데이터 조회
+              Optional<Meal> existingMealOptional = mealRepository.findBySchoolCodeAndMealDate(
+                  school, mealDate);
+
+              if (existingMealOptional.isPresent()) {
+                // 기존 데이터가 있으면 업데이트
+                Meal existingMeal = existingMealOptional.get();
+                existingMeal.setMealNames(mealNames);
+                mealRepository.save(existingMeal);
+              } else {
+                // 기존 데이터가 없으면 새로 추가
+                Meal meal = new Meal();
+                meal.setSchoolCode(school);
+                meal.setMealNames(mealNames);
+                meal.setMealDate(mealDate);
+                mealRepository.save(meal);
+              }
```
이렇게 해보셔도 좋을 것 같아요. Optional type을 변수로 선언하는 것은 안티패턴이라 피하시면 좋을 것 같습니다
```suggestion
              mealRepository.findBySchoolCodeAndMealDate(school, mealDate).ifPresent(existMeal -> 
                  { 
                       existMeal.setMealNames(mealNames);
                       mealRepository.save(existMeal);
                   }).orElseGet(() -> {
                      // 기존 데이터가 없으면 새로 추가
                      Meal meal = new Meal();
                      meal.setSchoolCode(school);
                      meal.setMealNames(mealNames);
                      meal.setMealDate(mealDate);
                      mealRepository.save(meal);
                    });
```

Hue-Jo replied:
@mingming-mentor 
멘토님께서 의도하셨던 건 .map().orElseGet() 이었겠지요..?
ifPresentOrElse()를 활용하여 수정하였는데 괜찮을까요?  
Optional이 편한 줄 알았는데 꽤나 금지하는 것들이 많네요. 덕분에 여러모로 알아갑니다!

mingming-mentor commented on SchoolMealBoard/src/main/java/com/zerobase/schoolmealboard/service/impl/MealServiceImpl.java:
```diff
@@ -53,75 +53,97 @@ public void fetchAndSaveMealInfo() {
+    LocalDate twoMonthsAgoEndDate = twoMonthsAgoStartDate.withDayOfMonth(twoMonthsAgoStartDate.lengthOfMonth()); // 두 달전의 마지막 날
+
+    // 두 달 전 데이터 삭제
+    mealRepository.deleteByMealDateBetween(twoMonthsAgoStartDate, twoMonthsAgoEndDate);
```
JPA로 delete를 부르시면 select가 무조건 발생하게 됩니다. entityManager로 해당 entity를 가지고 와야하니깐요.
현재 구조상으로는 굳이 select가 필요한 쿼리는 아니라고 보입니다.
따라서 native query로 작성해주시면 좋을 것 같아요

Hue-Jo replied:
@mingming-mentor select 을 하면 불필요한 데이터들까지 읽게 되어서 성능이 저하되는 걸 피하기 위함일까요?

mingming-mentor replied:
@Hue-Jo 님 데이터를 가지고 올 필요가 없는데 가지고 오는 것이 성능 저하를 일으키기 때문입니다. 
현재 코드상으로는 두달전 데이터를 select 한다음에 delete를 할 이유가 없어보여서요

mingming-mentor reviewed (approved):
수고 많으셨어요. 500에러가 뜨는 것에 대해서는 에러관련하여 메세지등을 첨부하여 슬랙으로 문의주시면 함께 원인을 찾아볼 수 있을 것 같습니다

Hue-Jo committed:
Refactor And Fix

1. 스프링배치 관련 설정 (실행이 안 되는 이유가 오타 때문이었음)
yml 파일에
batch:
    job:
      enabled: false
      names: fetchAndSaveMealJob,deleteOldMealJob 추가
2. native query 작성
3. 데이터베이스에 저장 관련
Optional 타입의 변수 선언하는 것을 피하여 리팩토링

Hue-Jo committed:
Create SignUp

회원가입 코드 구현
http 테스트 완 (유저 테이블에 데이터 들어감 )

Hue-Jo committed:
Create GlobalExceptionHandler

유효성 검증을 위한 핸들러
